packages/framework/src/Model/Order/Order.php

```diff
@@ -932,4 +942,38 @@ public function calculateTotalPrice(OrderPriceCalculation $orderPriceCalculation
         $orderTotalPrice = $orderPriceCalculation->getOrderTotalPrice($this);
         $this->setTotalPrice($orderTotalPrice);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price $productPrice
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderProductFactoryInterface $orderProductFactory
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\Order\OrderPriceCalculation $orderPriceCalculation
+     * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderProduct
+     */
+    public function createOrderProduct(
```

Comment:
You made me thinking about the use case, and I have another suggestion - let's rename it to `Order::addProduct`, because that is what the use case does... what do you think?

Reply:
And I think this logic belongs to the `Order`

Reply:
yes, I like it :slightly_smiling_face: it is funny, the controller action is called `addProductAction()`, but then you have `OrderItemFacade::createOrderProductInOrder()` and `Order::createOrderProduct()` - I believe both of them should be renamed :+1: